lib/internal/fs/utils.js

```diff
@@ -709,14 +709,38 @@ const validateOffsetLengthWrite = hideStackFrames(
   },
 );
 
-const validatePath = hideStackFrames((path, propName = 'path') => {
+/**
+ * Validates the given path based on the expected type (file or directory).
+ *
+ * @param {string} path - The path to validate.
+ * @param {string} [propName='path'] - The name of the property being validated.
+ * @param {boolean} expectFile - If true, expects the path to be a file.
+ * @throws {TypeError} Throws if the path is not a string, Uint8Array, or URL without null bytes.
+ * @throws {Error} Throws if the path does not meet the expected type (file).
+ */
+const validatePath = hideStackFrames((path, propName = 'path', expectFile) => {
```

Comment:
maybe we should go for `options` instead of `expectFile` and pass `{ expectFile: true, syscall: 'cp' }`?

Reply:
And just check for `options != null` in 729?

Reply:
Fixed.